cloudinit/sources/helpers/azure.py

```diff
@@ -239,6 +240,15 @@ def push_log_to_kvp(file_name=CFG_BUILTIN['def_log_file']):
             "Exception when dumping log file: %s" % repr(ex),
             logger_func=LOG.warning)
 
+    LOG.debug("Dumping dmesg log to KVP")
+    try:
+        out, _ = subp.subp(['dmesg'], decode=False, capture=True)
```

Comment:
Thanks for this @anhvoms . Could we get some semblance of unit test coverage for this path please.

Maybe we could have a test in tests/unittests/test_reporting_hyperv.py  confirming that regardless of what "log" is being written that dmesg is also being performed? 

I'd also really appreciate if we can get a test there that raised an error from report_compressed_event during push_log_to_kvp (to assert Exception handling behavior).

Reply:
Thanks for the review @blackboxsw . I have added unit tests to cover the angles you mentioned.